src/test/java/data/exceptions/MarkTaskExceptionTest.java

```diff
@@ -0,0 +1,95 @@
+package data.exceptions;
+
+import data.Task;
+import data.TaskManager;
+import data.TaskType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static data.TaskManager.addTask;
+import static data.TaskManager.deleteAllTasksOnDate;
+import static data.TaskManager.getDayTasks;
+import static data.exceptions.MarkTaskException.checkIfTaskIndexIsValidForMarkingTask;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class MarkTaskExceptionTest {
+
+    private TaskManager taskManager;
+
+    @BeforeEach
+    void setUp() {
+        taskManager = new TaskManager();
+    }
+
+    @AfterEach
+    void resetTaskManager() {
+        LocalDate date = LocalDate.now();
+        deleteAllTasksOnDate(taskManager, date);
+    }
+
+    @Test
+    public void checkIfTaskIndexIsValidForMarkingTask_validIndexGivenForDayWithTask_noExceptionThrown ()
+            throws TaskManagerException {
+
+        // Arrange
+        LocalDate date = LocalDate.now();
+        String taskDescription = "Test Todo";
+
+        // Act
+        TaskType testTaskType = TaskType.TODO;
+        String[] dummyTestDates = new String[]{null};
+        String[] dummyTestTimes = new String[]{null};
+        addTask(date, taskDescription, testTaskType, dummyTestDates, dummyTestTimes);
+        List<Task> dayTasks = getDayTasks(date);
+
+        // Assert
+        assertDoesNotThrow(() -> checkIfTaskIndexIsValidForMarkingTask(dayTasks, 1));
+
+    }
```

Comment:
Testing all possible cases, even where exceptions are thrown and not, is great for making our code robust!